hoomd/md/EvaluatorPairTable.h

```diff
@@ -0,0 +1,236 @@
+    [rmin,rmax], inclusive.
+
+    \b Interpolation
+    Values are interpolated linearly between two points straddling the given r. For a given r, the first point needed, i
+    can be calculated via i = floorf((r - rmin) / dr). The fraction between ri and ri+1 can be calculated via
+    f = (r - rmin) / dr - Scalar(i). And the linear interpolation can then be performed via V(r) ~= Vi + f * (Vi+1 - Vi)
+    \ingroup computes
+*/
+class EvaluatorPairTable
+    {
+    public:
+        //! Define the parameter type used by this pair potential evaluator
+        struct param_type
+            {
+            unsigned int width;
+            Scalar rmin;
+            ManagedArray<Scalar> V_table;
+            ManagedArray<Scalar> F_table;
+
+            #ifdef ENABLE_HIP
+            //! Set CUDA memory hints
+            void set_memory_hint() const
+                {
+                // default implementation does nothing
+                }
+            #endif
+
+            #ifndef __HIPCC__
+            param_type() : width(0), rmin(0.0), V_table({}), F_table({}) {}
+
+            param_type(pybind11::dict v)
+                {
+                width = v["width"].cast<unsigned int>();
+                rmin = v["r_min"].cast<Scalar>();
+                unsigned int align_size = 8; //for AVX
+                unsigned int N_align =((width + align_size - 1)/align_size)*align_size;
+                V_table = ManagedArray<Scalar>(N_align, false, 32); // 32byte alignment for AVX
+                F_table = ManagedArray<Scalar>(N_align, false, 32);
+                auto V_py = v["V"].cast<pybind11::array_t<Scalar>>().unchecked<1>();
+                auto F_py = v["F"].cast<pybind11::array_t<Scalar>>().unchecked<1>();
+                for (unsigned int i = 0; i < width; i++)
+                    {
+                    V_table[i] = V_py[i];
+                    F_table[i] = F_py[i];
+                    }
+                if (V_table.size() != F_table.size())
+                    {throw std::runtime_error("The length of V and F arrays must be equal");}
+                }
+
+            pybind11::dict asDict()
+                {
+                Scalar *V = new Scalar[width];
+                Scalar *F = new Scalar[width];
+                for (unsigned int i = 0; i < width; i++)
+                    {
+                    V[i] = (Scalar) V_table[i];
+                    F[i] = (Scalar) F_table[i];
+                    }
+
+                pybind11::capsule free_V(V, [](void *f)
+                    {
+                    Scalar *V = reinterpret_cast<Scalar *>(f);
+                    delete[] V;
+                    });
+                pybind11::capsule free_F(F, [](void *f)
+                    {
+                    Scalar *F = reinterpret_cast<Scalar *>(f);
+                    delete[] F;
+                    });
+
+                pybind11::dict v;
+                v["V"] = pybind11::array_t<Scalar>(pybind11::array::ShapeContainer({width,}),
```

Comment:
@b-butler 

Are numpy arrays pickleable? Do we need to make changes here so that table potentials will work with the new pickling infrastructure?

Reply:
NumPy arrays are pickable.